File: Tugas1/CPP/testCircle.cpp
#include <iostream>
#include "Circle.h"
using namespace std;

int main(){
    // Test semua konstruktor dan toString
    /* Radius dan warna tertentu */
    Circle c1(1.1,"biru");
    cout << c1.toString() << endl;

    /* Radius tertentu warna default */
    Circle c2(2.2);
    cout << c2.toString() << endl;

    /* Radius dan warna default */
    Circle c3;
    cout << c3.toString() << endl;

    // Test getter/setter
    /* test setter */
    c1.setelRadius(3.3);
    c1.setelWarna("hijau");
    /* menggunakan string untuk mengecek instance yang sudah dimodifikasi */
    cout << c1.toString() << endl;
    /* test getter */
    cout << "Radius : " << c1.ambilRadius() << endl;
    cout << "Warna : " << c1.ambilWarna() << endl;

    // Test Luas dan keliling
    /* Luas */
    cout << "Luas : " << c1.hitungLuas() << endl;
    /* Keliling */
    cout << "Keliling : " << c1.hitungKeliling() << endl;

    return 0;
}

File: Tugas1/Java/TestCircle.java
/**
 * Test untuk kelas Circle
 */
public class TestCircle {
   public static void main(String[] args) {
      // Test semua kontrukstor dan toString()
      Circle c1 = new Circle(1.1, "biru");
      System.out.println(c1); // jalankan toString() secara implisit
      // Circle[radius=1.1, warna=biru]
      Circle c2 = new Circle(2.2);
      System.out.println(c2);
      // Circle[radius=2.2, warna=merah]
      Circle c3 = new Circle();
      System.out.println(c3);
      // Circle[radius=1.0, warna=merah]

      // Tes untuk mengatur dan mengambil
      c1.setRadius(3.3);
      c1.setWarna("hijau");
      System.out.println(c1); // gunakan toString() untuk memeriksa instance yang dimodifikasi
      // Circle[radius=3.3, warna=hijau]
      System.out.println("Radius : " + c1.getRadius());
      // Radius : 3.3
      System.out.println("Warna : " + c1.getWarna());
      // Warna : hijau

      // Test getLuas() dan getKeliling()
      System.out.printf("Luas : %.2f%n", c1.getLuas());
      // Luas : 34.21
      System.out.printf("Kurikulum : %.2f%n", c1.getKeliling());
      // Keliling : 20.73
   }
}